### PR TITLE
Migrate the emoji sample app to use Redwood Layout Row/Column. 

### PR DESCRIPTION
**Pull request by colinrtwhite**
Known issues:

- The UI doesn't fill the full width. This occurs on both platforms.
- Padding isn't uniform between both platforms. We'll need to translate the common padding values to proportionally sized values on each platform.
- The scrolling column doesn't scroll to the end of all items. This occurs on both platforms.
- iOS sometimes ellipses the text after filtering the list.

Demos:

[device-2022-11-07-023756.webm](https://user-images.githubusercontent.com/5580160/200290739-780d02d3-3022-47f9-8661-5152fc06cc7c.webm)

https://user-images.githubusercontent.com/5580160/200290757-4533cce3-a30b-4750-9ce2-18f226a5c1cb.mp4

### CODE REVIEW

File: redwood-layout-composeui/src/main/kotlin/app/cash/redwood/layout/composeui/ComposeRow.kt [hunk 1]
```diff
@@ -26,7 +26,7 @@ import app.cash.redwood.layout.widget.Row
 import app.cash.redwood.widget.Widget
 
 internal class ComposeRow : Row<@Composable () -> Unit> {
-  private val container = ComposeFlexContainer(FlexDirection.Column)
+  private val container = ComposeFlexContainer(FlexDirection.Row)
```
Comment (JakeWharton) [inline redwood-layout-composeui/src/main/kotlin/app/cash/redwood/layout/composeui/ComposeRow.kt:29]: Haha whoops

File: samples/emoji-search/android/src/main/java/app/cash/zipline/samples/emojisearch/AndroidEmojiSearchWidgetFactory.kt [hunk 2]
```diff
@@ -16,12 +16,12 @@
...
 import androidx.compose.runtime.Composable
+import app.cash.redwood.layout.composeui.ComposeRedwoodLayoutWidgetFactory
+import app.cash.redwood.layout.widget.RedwoodLayoutWidgetFactory
 import example.schema.widget.EmojiSearchWidgetFactory
 
 object AndroidEmojiSearchWidgetFactory : EmojiSearchWidgetFactory<@Composable () -> Unit> {
-  override fun Row() = ComposeUiRow()
-  override fun Column() = ComposeUiColumn()
-  override fun ScrollableColumn() = ComposeUiScrollableColumn()
+  override val RedwoodLayout = ComposeRedwoodLayoutWidgetFactory()
```
Comment (JakeWharton) [inline samples/emoji-search/android/src/main/java/app/cash/zipline/samples/emojisearch/AndroidEmojiSearchWidgetFactory.kt:24]: Hell yeah

**Review by JakeWharton — COMMENTED** [review]
Good so far

File: redwood-layout-composeui/src/main/kotlin/app/cash/redwood/layout/composeui/ComposeRow.kt (hunk 1 above)
Comment (colinrtwhite) [reply redwood-layout-composeui/src/main/kotlin/app/cash/redwood/layout/composeui/ComposeRow.kt:29]: Yep 🤦

**Comment by colinrtwhite** [comment]
Okay this should be good to review now. It's definitely not production ready, but both samples run with only a couple (common) issues.

**Review by JakeWharton — APPROVED** [review]

**Pull request merged**